inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/pulsar/PulsarMultiTopicsFetcher.java

```diff
@@ -0,0 +1,432 @@
+                    break;
+                }
+                String offsetKey = getOffset(msg.getMessageId());
+                currentConsumer.put(offsetKey, topic, msg.getMessageId());
+
+                //deserialize
+                List<InLongMessage> inLongMessages = deserializer
+                        .deserialize(context, topic, msg.getProperties(), msg.getData());
+                // intercept
+                inLongMessages = interceptor.intercept(inLongMessages);
+                if (inLongMessages.isEmpty()) {
+                    ack(offsetKey);
+                    continue;
+                }
+
+                msgs.add(new MessageRecord(topic.getTopicKey(),
+                        inLongMessages,
+                        offsetKey, System.currentTimeMillis()));
+                context.getStateCounterByTopic(topic).addConsumeSize(msg.getData().length);
+            }
+            context.getDefaultStateCounter().addMsgCount(msgs.size());
+            return msgs;
+        }
+
+        @Override
+        public void run() {
+            boolean hasPermit;
+            while (true) {
+                hasPermit = false;
+                try {
+                    if (context.getConfig().isStopConsume() || stopConsume) {
+                        TimeUnit.MILLISECONDS.sleep(50);
+                        continue;
+                    }
+
+                    if (sleepTime > 0) {
+                        TimeUnit.MILLISECONDS.sleep(sleepTime);
+                    }
+
+                    context.acquireRequestPermit();
+                    hasPermit = true;
+                    context.getDefaultStateCounter().addMsgCount(1L).addFetchTimes(1L);
+
+                    long startFetchTime = System.currentTimeMillis();
+                    Messages<byte[]> pulsarMessages = currentConsumer.batchReceive();
+
+                    context.getDefaultStateCounter().addFetchTimeCost(System.currentTimeMillis() - startFetchTime);
+                    if (null != pulsarMessages && pulsarMessages.size() != 0) {
+                        List<MessageRecord> msgs = this.processPulsarMsg(pulsarMessages);
+                        handleAndCallbackMsg(msgs);
+                        sleepTime = 0L;
+                    } else {
+                        context.getDefaultStateCounter().addEmptyFetchTimes(1L);
+                        emptyFetchTimes++;
+                        if (emptyFetchTimes >= context.getConfig().getEmptyPollTimes()) {
+                            sleepTime = Math.min((sleepTime += context.getConfig().getEmptyPollSleepStepMs()),
+                                    context.getConfig().getMaxEmptyPollSleepMs());
+                            emptyFetchTimes = 0;
+                        }
+                    }
+                } catch (Exception e) {
+                    context.getDefaultStateCounter().addFetchErrorTimes(1L);
+                    LOGGER.error("failed to fetch msg: {}", e.getMessage(), e);
```

Comment:
Just change to `LOGGER.error("failed to fetch msg: ", e);`.

Reply:
fixed, thx